src/fsharp/IlxGen.fs

```diff
@@ -1139,12 +1146,18 @@ and TypeDefsBuilder() =
 
     member b.AddTypeDef(tdef:ILTypeDef, eliminateIfEmpty, addAtEnd) = 
         let idx = if addAtEnd then (countDown <- countDown - 1; countDown) else tdefs.Count
-        tdefs.Add (tdef.Name, (idx, (new TypeDefBuilder(tdef), eliminateIfEmpty)))
+        let pt =
+            deferredProvidedTypes
```

Comment:
Can you index defferredProvidedTypes to avoid this potentially quadratic lookup?